hugr-model/src/v0/mod.rs

```diff
@@ -0,0 +1,621 @@
+    pub name: &'a str,
+    /// The static parameters of the alias.
+    pub params: &'a [Param<'a>],
+    /// The type of the alias.
+    pub r#type: TermId,
+}
+
+/// A metadata item.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub struct MetaItem<'a> {
+    /// Name of the metadata item.
+    pub name: &'a str,
+    /// Value of the metadata item.
+    pub value: TermId,
+}
+
+/// A reference to a global variable.
+///
+/// Global variables are defined in nodes.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub enum GlobalRef<'a> {
+    /// Reference to the global that is defined by the given node.
+    Direct(NodeId),
+    /// Reference to the global with the given name.
+    Named(&'a str),
+}
+
+impl std::fmt::Display for GlobalRef<'_> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            GlobalRef::Direct(id) => write!(f, ":{}", id.index()),
+            GlobalRef::Named(name) => write!(f, "{}", name),
+        }
+    }
+}
+
+/// A reference to a local variable.
+///
+/// Local variables are defined as parameters to nodes.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub enum LocalRef<'a> {
+    /// Reference to the local variable by its parameter index.
+    Index(u16),
+    /// Reference to the local variable by its name.
+    Named(&'a str),
+}
+
+impl std::fmt::Display for LocalRef<'_> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            LocalRef::Index(index) => write!(f, "?:{}", index),
+            LocalRef::Named(name) => write!(f, "?{}", name),
+        }
+    }
+}
+
+/// A reference to a link.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub enum LinkRef<'a> {
+    /// Reference to the link by its id.
+    Id(LinkId),
+    /// Reference to the link by its name.
+    Named(&'a str),
+}
+
+impl std::fmt::Display for LinkRef<'_> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            LinkRef::Id(id) => write!(f, "%:{})", id.index()),
+            LinkRef::Named(name) => write!(f, "%{}", name),
+        }
+    }
+}
+
+/// A term in the compile time meta language.
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub enum Term<'a> {
+    /// Standin for any term.
+    Wildcard,
+
+    /// The type of runtime types.
+    ///
+    /// `type : static`
+    Type,
+
+    /// The type of static types.
+    ///
+    /// `static : static`
+    StaticType,
+
+    /// The type of constraints.
+    ///
+    /// `constraint : static`
+    Constraint,
+
+    /// A local variable.
+    Var(LocalRef<'a>),
+
+    /// A symbolic function application.
+    ///
+    /// `(GLOBAL ARG-0 ... ARG-n)`
+    Apply {
+        // TODO: Should the name be replaced with the id of the node that defines
+        // the function to be applied? This could be a type, alias or function.
```

Comment:
seems like this todo has been done by using a ref, should `name` field be renamed?

Reply:
Yes, good catch. Do you have an idea for a good name? In my draft for the capnp serialization I've used `global`, but I'm not entirely sure about that.

Reply:
`func`, `term`?